module/Entra/Microsoft.Entra/Users/Set-EntraCBACertificateUserId.ps1

```diff
@@ -0,0 +1,66 @@
+function  Set-EntraCBACertificateUserId {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$UserId,
+        [Parameter(Mandatory = $false)]    
+        [string]$CertPath,
+        [Parameter(Mandatory = $false)]
+        [System.Security.Cryptography.X509Certificates.X509Certificate2]$Cert,
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("PrincipalName", "RFC822Name", "IssuerAndSubject", "Subject", "SKI", "SHA1PublicKey", "IssuerAndSerialNumber")]
+        [string[]]$CertificateMapping
+    )
+
+    PROCESS {
+        # Ensure all provided parameters are bound correctly
+        $params = @{}
+        foreach ($paramName in $PSBoundParameters.Keys) {
+            $params[$paramName] = $PSBoundParameters[$paramName]
+        }
+        # Ensure at least one of the parameters about certificate is provided
+        if (-not $Cert -and -not $CertPath) {
+            throw "You must provide either -Cert or -CertPath."
+        }
+        . "$PSScriptRoot/../Utilities/Get-EntraUserCertificateUserIdsFromCertificate.ps1"
```

Comment:
Will this linking work when we ship this to PowerShell gallery?

If this cmdlet has a dependency on another cmdlet in the `Utilities` submodule, why don't we move this to `Utilities`?

For performance reasons, I would not want `Utilities` sub-module to be a required module in `Users` sub-module

Reply:
`Set-EntraUserCBACertificateUserId` utilizes MsGraph to modify the user, making it logical for it to be included in the Users module.

On the other hand, `Get-EntraUserCertificateUserIdsFromCertificate` does not alter the Entra object via MsGraph. In the future, we might have more scripts like this. So, these "stand alone" scripts are more suitable for the utilities module. 

I think @SteveMutungi254  does agree that makes sense this organization.
Otherwise I would suggest to create a "CertificateBasedAuthentication` module instead of `Utilities` which will contain both `Get-EntraUserCertificateUserIdsFromCertificate` and `Set-EntraUserCBACertificateUserId`